### PR TITLE
ci: use images from quay.io

### PR DESCRIPTION
**Pull request by nebril**
This change causes jenkins CI jobs to not build images and instead use
images built by PR github action

related: https://github.com/cilium/cilium/issues/14928

### CODE REVIEW

**Comment by nebril** [comment]
test-net-next

**Comment by nebril** [comment]
test-net-next

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 1]
```diff
@@ -193,6 +174,19 @@ pipeline {
...
         }
+        stage ("Wait for images") {
+            options {
+                timeout(time: 10, unit: 'MINUTES')
+            }
+            steps {
+                retry(15) {
+                    sleep(time: 60)
+                    sh 'curl --silent -f -lSL "https://quay.io/api/v1/repository/cilium/cilium-ci/tag/${ghprbActualCommit}/images"'
+                    sh 'curl --silent -f -lSL "https://quay.io/api/v1/repository/cilium/cilium-operator-ci/tag/${ghprbActualCommit}/images"'
```
Comment (aanm) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile]: s/`cilium-operator-ci`/`operator-ci`

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 2]
```diff
@@ -225,21 +219,12 @@ pipeline {
...
-                        )}"""
-                CILIUM_OPERATOR_TAG = "latest"
-                HUBBLE_RELAY_IMAGE= """${sh(
-                        returnStdout: true,
-                        script: 'echo -n $(${TESTDIR}/print-node-ip.sh)/cilium/hubble-relay'
-                        )}"""
-                HUBBLE_RELAY_TAG = "latest"
+                CILIUM_IMAGE = "quay.io/cilium/cilium-ci"
+                CILIUM_TAG = "${ghprbActualCommit}"
+                CILIUM_OPERATOR_IMAGE= "quay.io/cilium/cilium-operator-ci"
```
Comment (aanm) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile]: same here

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by nebril** [comment]
test-net-next

**Comment by nebril** [comment]
test-net-next

**Comment by nebril** [comment]
Blocked by https://github.com/cilium/cilium/pull/14952

**Comment by nebril** [comment]
Also blocked by race detection images

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-net-next

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-4.19

**Comment by nebril** [comment]
test-4.9

**Comment by nebril** [comment]
test-runtime

**Comment by nebril** [comment]
test-gke

File: jenkinsfiles/ginkgo-gke.Jenkinsfile [hunk 3]
```diff
@@ -63,11 +63,6 @@ pipeline {
                 sh 'ls -A | grep -v src | xargs mv -t ${PROJ_PATH}'
             }
         }
-        stage('Start docker registry for build') {
-            steps{
-                sh 'cd ${TESTDIR}/gke; ./start-registry.sh'
```
Comment (tklauser) [inline jenkinsfiles/ginkgo-gke.Jenkinsfile:68]: It looks like the `start-registry.sh` script is no longer used after this PR. Should it be removed as well?

File: jenkinsfiles/ginkgo-gke.Jenkinsfile [hunk 4]
```diff
@@ -199,7 +182,6 @@ pipeline {
   }
     post {
         always {
-            sh 'cd ${TESTDIR}/gke; ./stop-registry.sh || true'
```
Comment (tklauser) [inline jenkinsfiles/ginkgo-gke.Jenkinsfile:202]: Same here, it looks like `stop-registry.sh` is now no longer used. Remove it as well?

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Review by aanm — APPROVED** [review]

**Comment by nebril** [comment]
test-gke

File: jenkinsfiles/ginkgo-gke.Jenkinsfile [hunk 5]
```diff
@@ -78,26 +73,32 @@ pipeline {
...
         }
-        stage('Make Cilium images and prepare gke cluster') {
+        stage('Wait for Cilium images and prepare gke cluster') {
             parallel {
-                stage('Make Cilium images') {
-                    steps {
-                        sh 'cd ${TESTDIR}; ./make-images-push-to-local-registry.sh $(gke/registry-ip.sh) ${TAG} "--no-cache"'
+                stage('Wait for images') {
+                    options {
+                        timeout(time: 10, unit: 'MINUTES')
```
Comment (aanm) [inline jenkinsfiles/ginkgo-gke.Jenkinsfile]: The images with `-race` might take longer time before they are available so this timeout needs to be bumped to at least 20 minutes.

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 6]
```diff
@@ -193,6 +175,19 @@ pipeline {
                 }
             }
         }
+        stage ("Wait for images") {
+            options {
+                timeout(time: 10, unit: 'MINUTES')
```
Comment (aanm) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile]: same here

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 7]
```diff
@@ -193,6 +175,19 @@ pipeline {
...
+            }
+            steps {
+                retry(15) {
```
Comment (aanm) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile]: retry 20

File: jenkinsfiles/ginkgo-gke.Jenkinsfile [hunk 8]
```diff
@@ -78,26 +73,32 @@ pipeline {
...
                     }
-                    post {
-                        unsuccessful {
-                            script {
-                                if  (!currentBuild.displayName.contains('fail')) {
-                                    currentBuild.displayName = 'building or pushing Cilium images failed ' + currentBuild.displayName
-                                }
-                            }
+                    steps {
+                        retry(15) {
```
Comment (aanm) [inline jenkinsfiles/ginkgo-gke.Jenkinsfile]: retry 20

**Review by aanm — CHANGES_REQUESTED** [review]

File: jenkinsfiles/ginkgo-gke.Jenkinsfile (hunk 5 above)
Comment (nebril) [reply jenkinsfiles/ginkgo-gke.Jenkinsfile]: Good catch.

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-race-4.9

**Review by aanm — APPROVED** [review]

**Pull request merged**